qcodes/utils/dataset/doNd.py

```diff
@@ -495,6 +496,242 @@ def do2d(
+            this may give a significant speedup.
+        additional_setpoints: A list of setpoint parameters to be registered in
+            the measurement but not scanned.
+    """
+    if do_plot is None:
+        do_plot = config.dataset.dond_plot
+    if show_progress is None:
+        show_progress = config.dataset.dond_show_progress
+
+    meas = Measurement(name=measurement_name, exp=exp)
+
+    def _make_nested_setpoints(
+        *params: Union[_AbstractSweep, ParamMeasT],
+    ) -> Union[np.ndarray, List[Tuple[()]]]:
+        """Create the cartesian product of all the setpoint values."""
+
+        setpoint_values: List[np.ndarray] = []
+        for param in params:
+            if isinstance(param, _AbstractSweep):
+                setpoint_values.append(param.get_setpoints())
+        setpoint_grids = np.meshgrid(*setpoint_values, indexing='ij')
+        flat_setpoint_grids = [np.ravel(grid, order='C')
+                               for grid in setpoint_grids]
+        flat_setpoints: Union[np.ndarray, List[Tuple[()]]]
+        if flat_setpoint_grids:
+            flat_setpoints = np.vstack(flat_setpoint_grids).T
+        else:
+            flat_setpoints = [()]
+        return flat_setpoints
+
+    def _parse_dond_arguments(*params: Union[_AbstractSweep, ParamMeasT]
+                              ) -> Tuple[
+        List[_BaseParameter],
+        Dict[_BaseParameter, Dict[str, float]],
+        List[ParamMeasT]
+    ]:
+        """
+        Parse supplied arguments for a dond scan.
+        """
+        params_set: List[_BaseParameter] = []
+        params_delay: Dict[_BaseParameter, Dict[str, float]] = {}
+        params_meas: List[ParamMeasT] = []
+        for param in params:
+            if isinstance(param, _AbstractSweep):
+                params_set.append(param._param)
+                params_delay[param._param] = {}
+                params_delay[param._param]['delay'] = param._delay
+            elif isinstance(param, _BaseParameter):
+                params_meas.append(param)
+            else:
+                raise ValueError(f'Cannot proceed: {param} is not either a'
+                                 ' sweep instance or measurement parameter.'
+                                 ' Please check the supplied `dond`'
+                                 ' arguments.')
+        return params_set, params_delay, params_meas
+
+    params_set, params_delay, params_meas = _parse_dond_arguments(*params)
+    nested_setpoints = _make_nested_setpoints(*params)
+
+    all_setpoint_params = tuple(params_set) + tuple(
+            s for s in additional_setpoints)
+
+    measured_parameters = tuple(param for param in params_meas
+                                if isinstance(param, _BaseParameter))
+
+    num_points_params_set: List[int] = []
+    for par in params:
+        if isinstance(par, _AbstractSweep):
+            num_points_params_set.append(int(par._num_points))
```

Comment:
This indicates that that num_points should have a property on the Sweep interface

Reply:
Exactly